### PR TITLE
Feature/create login page

### PR DESCRIPTION
**Pull request by NIKOLENKO-TE**

### CODE REVIEW

File: src/assets/scss/elements/_login_register.scss [hunk 1]
```diff
@@ -0,0 +1,264 @@
+/* Login Register page css */
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_login_register.scss]: delete this line

File: src/assets/scss/elements/_login_register.scss [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+.container {
```
Comment (Igor-Klein-1988) [inline src/assets/scss/elements/_login_register.scss]: rename all classes or delete if you don't use

File: src/components/FloatingInput.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// FloatingInput.tsx
```
Comment (Igor-Klein-1988) [inline src/components/FloatingInput.tsx]: delete

File: src/components/LoginPage/LoginPage.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
+// LoginPage.tsx
```
Comment (Igor-Klein-1988) [inline src/components/LoginPage/LoginPage.tsx]: delete

File: src/components/LoginPage/LoginPage.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+import { Form, Button } from 'react-bootstrap';
+import { Link } from 'react-router-dom';
+import { Formik, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import Col from 'react-bootstrap/Col';
+import InputGroup from 'react-bootstrap/InputGroup';
+import { Icon } from 'react-icons-kit';
+import { eyeBlocked } from 'react-icons-kit/icomoon/eyeBlocked';
+import { eye } from 'react-icons-kit/icomoon/eye';
+import { useAppDispatch } from '../../app/hooks';
```
Comment (Igor-Klein-1988) [inline src/components/LoginPage/LoginPage.tsx]: add empty line

File: src/components/RegisterPage/RegisterPage.tsx [hunk 6]
```diff
@@ -0,0 +1,470 @@
+// RegisterPage.tsx
```
Comment (Igor-Klein-1988) [inline src/components/RegisterPage/RegisterPage.tsx]: delete

File: src/components/RegisterPage/RegisterPage.tsx [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+		</Form.Group>
+	);
+	const lastName = (
+		<Form.Group as={Col} id="formLastName">
+			<InputGroup>
+				<Field as={FloatingInput} id="lastName" name="lastName" placeholder={t('last_name')} />
+			</InputGroup>
+			<ErrorMessage name="lastName" component="div" className="validation_warning__message" />
+		</Form.Group>
+	);
```
Comment (Igor-Klein-1988) [inline src/components/RegisterPage/RegisterPage.tsx]: create componet for all  firstName, lastName add other

```
 newComponent (props) =>
 const {label, asFiels, id, name, placeholder, component, className} = props
```

File: src/components/RegisterPage/validationRegisterRules.tsx [hunk 8]
```diff
@@ -0,0 +1,652 @@
+// validationRegisterRules.tsx
```
Comment (Igor-Klein-1988) [inline src/components/RegisterPage/validationRegisterRules.tsx]: delete

File: src/features/auth/api.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
+// api.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/api.ts]: delete

File: src/features/auth/authSlice.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
+// src/features/auth/authSlice.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/authSlice.ts]: delete

File: src/features/auth/regSlice.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
+// authSlice.ts
+
+// src/features/auth/regSlice.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/regSlice.ts]: delete

File: src/features/auth/types/AuthState.ts [hunk 12]
```diff
@@ -0,0 +1,5 @@
+// AuthState.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/AuthState.ts]: delete

File: src/features/auth/types/LoginCredentials.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+// LoginCredentials.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/LoginCredentials.ts]: delete

File: src/features/auth/types/RegState.ts [hunk 14]
```diff
@@ -0,0 +1,6 @@
+// RegState.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/RegState.ts]: delete

File: src/features/auth/types/RegisterCredentials.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
+// RegisterCredentials.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/RegisterCredentials.ts]: delete

File: src/features/auth/types/User.ts [hunk 16]
```diff
@@ -0,0 +1,6 @@
+// User.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/User.ts]: delete

File: src/features/auth/types/newUser.ts [hunk 17]
```diff
@@ -0,0 +1,15 @@
+// NewUser.ts
```
Comment (Igor-Klein-1988) [inline src/features/auth/types/newUser.ts]: delete

File: src/index.scss [hunk 18]
```diff
@@ -45,6 +45,8 @@
 @import './assets/scss/elements/change_password';
 @import './assets/scss/elements/my_auctions';
 @import './assets/scss/elements/timer';
+@import './assets/scss/elements/login_register';
+// @import "elements/header";
```
Comment (Igor-Klein-1988) [inline src/index.scss]: delete

File: src/components/LoginPage/LoginPage.tsx [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+	const appDispatch = useAppDispatch();
+	const [passwordShow, setPasswordShow] = useState(false);
+	const [passwordIcon, setPasswordIcon] = useState(eye);
+	const [message] = useState<string>('');
+	const handleLogin = (formValues: { email: string; password: string }): void => {
+		appDispatch(login(formValues))
+			.then((res) => {
+				const payload = res.payload as { message: string };
+				if (payload.message) {
+					// Дальнейшие действия после успешной регистрации
```
Comment (maiorovvv) [inline src/components/LoginPage/LoginPage.tsx]: please, remove this comment

File: src/components/LoginPage/LoginPage.tsx [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+				}
+			})
+			.catch(() => {
+				// Обработка ошибки при регистрации
```
Comment (maiorovvv) [inline src/components/LoginPage/LoginPage.tsx]: please, remove this comment

File: src/assets/scss/elements/_login_register.scss [hunk 21]
```diff
@@ -0,0 +1,264 @@
...
+	}
+}
+
+// .login__account--input::placeholder {
+// 	font-size: 1.5rem;
+// 	margin-top: -5px;
+// 	background: transparent;
+// 	background-color: transparent !important;
+// 	margin-top: 5px;
+// }
```
Comment (maiorovvv) [inline src/assets/scss/elements/_login_register.scss]: please, remove this comment

File: src/app/hooks.ts [hunk 22]
```diff
@@ -1,6 +1,6 @@
+// hooks.ts
```
Comment (maiorovvv) [inline src/app/hooks.ts]: please, remove this comment

**Review by maiorovvv — CHANGES_REQUESTED** [review]

File: src/components/FloatingInput.tsx [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+		}
+	}, [isInvalid]);
+
+	const handleBlur = (): void => {
+		formik.setFieldTouched(name, true);
+	};
+
+	return (
+		<div
+			style={{ justifyTracks: 'center', marginBottom: `1rem`, paddingTop: `2rem` }}
```
Comment (DenysBabenko29) [inline src/components/FloatingInput.tsx]: can you transfer it to scss?

File: src/components/FloatingInput.tsx [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+				name={name}
+				style={{ height: `4.8rem`, paddingTop: `33px`, fontSize: `20px`, paddingBottom: `15px` }}
+				placeholder={placeholder}
+				onFocus={handleBlur}
+				onBlur={handleBlur}
+			/>
+
+			<label
+				htmlFor={id}
+				style={{
```
Comment (DenysBabenko29) [inline src/components/FloatingInput.tsx]: can you transfer it to ssss?

**Review by DenysBabenko29 — CHANGES_REQUESTED** [review]

File: src/components/FloatingInput.tsx (hunk 24 above)
Comment (NIKOLENKO-TE) [reply src/components/FloatingInput.tsx]: done

File: src/assets/scss/elements/_login_register.scss (hunk 2 above)
Comment (NIKOLENKO-TE) [reply src/assets/scss/elements/_login_register.scss]: done

File: src/components/RegisterPage/RegisterPage.tsx (hunk 7 above)
Comment (NIKOLENKO-TE) [reply src/components/RegisterPage/RegisterPage.tsx]: done

**Review by Igor-Klein-1988 — DISMISSED** [review]

File: src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx [hunk 25]
```diff
@@ -0,0 +1,65 @@
+import { FC, useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { Button } from 'react-bootstrap';
+
+const RestoreUserValidationWait: FC = (): JSX.Element => {
+	const { t } = useTranslation('LoginRegisterPage');
+	const navigate = useNavigate();
+	const [countdown, setCountdown] = useState(60);
```
Comment (DenysBabenko29) [inline src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx]: magic number, put it in constants

File: src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+
+	useEffect(() => {
+		const redirectTimeout = setTimeout(() => {
+			setCountdown((prevCountdown) => {
+				if (prevCountdown > 0) {
+					return prevCountdown - 1;
```
Comment (DenysBabenko29) [inline src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx]: same

File: src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+				} else {
+					navigate('/home');
+					return 0;
```
Comment (DenysBabenko29) [inline src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx]: same

File: src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+				}
+			});
+		}, 1000);
```
Comment (DenysBabenko29) [inline src/components/LoginRegisterForgotPassword/RestoreUserValidationWait.tsx]: same

**Review by DenysBabenko29 — CHANGES_REQUESTED** [review]

**Review by DenysBabenko29 — APPROVED** [review]

**Pull request merged**